tools/sigclient/pkg/ingest/ingest.go

```diff
@@ -151,16 +151,68 @@ func generateOpenTSDBBody(recs int, rdr utils.Generator) ([]byte, error) {
 	return retVal, nil
 }
 
-func runIngestion(iType IngestType, rdr utils.Generator, wg *sync.WaitGroup, url string, totalEvents int,
-	continous bool, batchSize, processNo int, indexPrefix string, ctr *uint64, bearerToken string,
-	indexName string, numIndices int) {
+var preGeneratedSeries []map[string]interface{}
+
+func generateUniqueID(series map[string]interface{}) string {
+	metricName := series["metric"].(string)
+	tags := series["tags"].(map[string]interface{})
+	var builder strings.Builder
+	builder.WriteString(metricName)
+	for key, value := range tags {
+		builder.WriteString(key)
+		builder.WriteString(fmt.Sprintf("%v", value))
+	}
+	return builder.String()
+}
+
+func generatePredefinedSeries(nMetrics int, cardinality uint64, gentype string) error {
+	preGeneratedSeries = make([]map[string]interface{}, 0, cardinality)
+	rdr, err := utils.InitMetricsGenerator(nMetrics, gentype)
+	if err != nil {
+		log.Errorf("generatePredefinedSeries: failed to initalize metrics generator: %+v", err)
+		return err
+	}
+	start := time.Now()
+	uniqueSeriesMap := make(map[string]struct{})
+	for uint64(len(preGeneratedSeries)) < cardinality {
+		currPayload, err := rdr.GetRawLog()
+		if err != nil {
+			log.Errorf("generatePredefinedSeries: failed to get raw log: %+v", err)
+			return err
+		}
+		uniqueID := generateUniqueID(currPayload)
+		if _, exists := uniqueSeriesMap[uniqueID]; !exists {
```

Comment:
you should not be doing this existence check here, the `generateUniqueID` function is suppose to generate a unique id, why don't you just do this check there ?

Reply:
so then this for loop becomes easy, it always knows it is going to get a uniqye id and all it has to do is insert in the array